ftw/news/tests/test_content_types.py

```diff
@@ -0,0 +1,50 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.news.testing import FTW_NEWS_FUNCTIONAL_TESTING
+from ftw.testbrowser import browsing
+from ftw.testbrowser.pages import factoriesmenu
+from plone.app.testing import setRoles
+from plone.app.testing import TEST_USER_ID
+from unittest2 import TestCase
+import transaction
+
+
+class TestContentTypes(TestCase):
+
+    layer = FTW_NEWS_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        super(TestContentTypes, self).setUp()
+        self.portal = self.layer['portal']
+        setRoles(self.portal, TEST_USER_ID, ['Manager', 'Site Administrator'])
```

Comment:
I would define a custom functional test case with a layer and helper methods, such as ``grant``, so that we can remove some redundancy.

I'd add it to the ``__init__.py`` of the ``tests`` package; something like that:

```python
from ftw.news.testing import FTW_NEWS_FUNCTIONAL_TESTING
from plone.app.testing import setRoles
from plone.app.testing import TEST_USER_ID
from unittest2 import TestCase
import transaction


class FunctionalTestCase(TestCase):
    layer = FTW_NEWS_FUNCTIONAL_TESTING

    def setUp(self):
        self.portal = self.layer['portal']

    def grant(self, *roles):
        setRoles(self.portal, TEST_USER_ID, list(roles))
        transaction.commit()
```

the tests can then be update:
```
from ftw.news.tests import FunctionalTestCase

class TestSomething(FunctionalTestCase):

    def setUp(self):
        super(....).setUp()
        self.grant('Manager', 'Site Administrator')
```

what do you think about that?